frontend/src/metabase/querying/components/FilterModal/TimeFilterEditor/TimeFilterEditor.unit.spec.tsx

```diff
@@ -0,0 +1,243 @@
+      userEvent.type(screen.getByPlaceholderText("Enter a time"), "10:20");
+      userEvent.click(document.body);
+
+      expect(getNextFilterName()).toBe("Time is before 10:20 AM");
+      expect(onInput).toHaveBeenCalled();
+    });
+
+    it("should add a filter with two values", async () => {
+      const { getNextFilterName, onInput } = setup({
+        query,
+        stageIndex,
+        column,
+      });
+
+      userEvent.click(screen.getByText("before"));
+      userEvent.click(await screen.findByText("Between"));
+      userEvent.clear(screen.getByPlaceholderText("Min"));
+      userEvent.type(screen.getByPlaceholderText("Min"), "10:15");
+      userEvent.clear(screen.getByPlaceholderText("Max"));
+      userEvent.type(screen.getByPlaceholderText("Max"), "20:40");
+      userEvent.click(document.body);
+
+      expect(getNextFilterName()).toBe("Time is 10:15 AM – 8:40 PM");
+      expect(onInput).toHaveBeenCalled();
+    });
+
+    it("should add a filter with no value", async () => {
+      const { getNextFilterName } = setup({
+        query,
+        stageIndex,
+        column,
+      });
+
+      userEvent.click(screen.getByText("before"));
+      userEvent.click(await screen.findByText("Is empty"));
+
+      expect(getNextFilterName()).toBe("Time is empty");
+    });
+
+    it("should not accept an empty string as a value", async () => {
+      const { getNextFilterName } = setup({
+        query,
+        stageIndex,
+        column,
+      });
+
+      userEvent.clear(screen.getByPlaceholderText("Enter a time"));
+      userEvent.click(document.body);
+
+      expect(getNextFilterName()).toBeNull();
+    });
+  });
+
+  describe("existing filter", () => {
+    it("should update a filter with one value", () => {
+      const { query, stageIndex, column, filter } = createQueryWithFilter({
+        operator: ">",
+        values: [new Date(2020, 0, 1, 10, 20)],
+      });
+      const { getNextFilterName } = setup({
+        query,
+        stageIndex,
+        column,
+        filter,
+      });
+
+      userEvent.clear(screen.getByDisplayValue("10:20"));
+      userEvent.type(screen.getByPlaceholderText("Enter a time"), "11:40");
+      userEvent.click(document.body);
+
+      expect(getNextFilterName()).toBe("Time is after 11:40 AM");
+    });
+
+    it("should update a filter with two values", () => {
+      const { query, stageIndex, column, filter } = createQueryWithFilter({
+        operator: "between",
+        values: [new Date(2020, 0, 1, 10, 20), new Date(2020, 0, 1, 12, 40)],
+      });
+      const { getNextFilterName } = setup({
+        query,
+        stageIndex,
+        column,
+        filter,
+      });
+
+      userEvent.clear(screen.getByDisplayValue("10:20"));
+      userEvent.type(screen.getByPlaceholderText("Min"), "11:40");
+      userEvent.clear(screen.getByDisplayValue("12:40"));
+      userEvent.type(screen.getByPlaceholderText("Max"), "15:10");
+      userEvent.click(document.body);
```

Comment:
if you need to trigger a blur event, there should be enough to use `userEvent.tab()`

<img width="765" alt="image" src="https://github.com/metabase/metabase/assets/125459446/b2c898ee-a0e8-4459-b6b1-ea5421ade712">